### PR TITLE
Feature/cc 0087 view court page in admin portal

### PR DESCRIPTION
**Pull request by EmilyGong111**

### CODE REVIEW

File: src/pages/courts/[courtID]/edit.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  const [centreCircleRadius, setCentreCircleRadius] = useState<number>();
+  const [threePointRadius, setThreePointRadius] = useState<number>();
+  const [threePointLine, setThreePointLine] = useState<number>();
+  const [lengthOfCorner, setLengthOfCorner] = useState<number>();
+  const [restrictedAreaLength, setRestrictedAreaLength] = useState<number>();
+  const [restrictedAreaWidth, setRestrictedAreaWidth] = useState<number>();
+  const [sideBorderWidth, setSideBorderWidth] = useState<number>();
+  const [lineBorderWidth, setLineBorderWidth] = useState<number>();
+  const [description, setDescription] = useState<string>();
+  const apiUrl = `${process.env.NEXT_PUBLIC_API_COURTS}/${CourtId}`;
```
Comment (Fuchih) [inline src/pages/courts/[courtID]/edit.tsx:34]: might need refactoring

File: src/components/SideBar/SideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <Grid
+        templateAreas={`"logo"
+                  "sideBarItem"
+                  "divider"
+                  " Profile"`}
+        gridTemplateRows={"10vh 70vh 2vh"}
+        fontWeight="bold"
+        margin="20px"
+      >
+        <GridItem pl="2" area={"logo"}>
```
Comment (Fuchih) [inline src/components/SideBar/SideBar.tsx:18]: pl --> paddingLeft

File: src/pages/courts/[courtID]/edit.tsx (hunk 1 above)
Comment (EmilyGong111) [reply src/pages/courts/[courtID]/edit.tsx:34]: Yes, on going! Thanks~

File: src/components/SideBar/components/Logo/Logo.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Text, Flex } from "@chakra-ui/react";
+import DashBoardLogo from "@/assets/svg/SideBarSvg/dashboard-log.svg";
+const Logo = () => {
```
Comment (Fuchih) [inline src/components/SideBar/components/Logo/Logo.tsx:3]: Add an empty line between L2 and L3 will be better

File: src/components/SideBar/components/SideBarItem/SideBarItem.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { Box, Icon } from "@chakra-ui/react";
+import Link from "next/link";
+import sideBarItemList from "./sideBarItemList";
+const SideBarItem: any = () => {
```
Comment (Fuchih) [inline src/components/SideBar/components/SideBarItem/SideBarItem.tsx:4]: Fix the any type if you can

Comment (EmilyGong111) [reply src/components/SideBar/components/SideBarItem/SideBarItem.tsx]: Done.

File: src/components/CourtsForm/CourtForm.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import { routeHandler } from "@/utils/routeHandler";
+import { Button, Flex, FormControl, FormLabel, Input } from "@chakra-ui/react";
+import { useState } from "react";
+import { api } from "../../utils/axios";
+import { ICourt } from "../../interfaces/courtData";
```
Comment (Fuchih) [inline src/components/CourtsForm/CourtForm.tsx:5]: Add Empty line

File: src/__tests__/components/SideBar/SideBar.test.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import SideBar from "../../../components/SideBar";
+import sidebarItemList from "@/components/SideBar/components/SideBarItem/sideBarItemList";
+import renderWithMockedProvider from "../../utils";
+describe("Sidebar component", () => {
```
Comment (Fuchih) [inline src/__tests__/components/SideBar/SideBar.test.tsx:5]: Add  an empty line

File: src/utils/routeHandler.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import router from "next/router";
+
+export const routeHandler = (endpoint?: string) => {
+  endpoint? router.push({
+    pathname: "courts/" + endpoint,
```
Comment (GitHubEason) [inline src/utils/routeHandler.ts:5]: Do you mind changing this part, making it more common?  make "courts/" a parameter.

File: src/components/CourtsForm/CourtForm.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+      <Flex maxWidth="1100px" flexDirection="row" flexWrap="wrap" gap="30px" alignSelf="center">
+        {Object.entries(inputData).map(([key, value]) => {
+          const headerCellContent = headerCellGenerator(key);
+          return (
+            <FormControl width="250px" key={key}>
+              <FormLabel htmlFor={key}>{headerCellContent}</FormLabel>
+              <Input
+                id={key}
+                value={value ? value.toString() : ""}
+                // placeholder={value ? value.toString() : ""}
```
Comment (Jzeva) [inline src/components/CourtsForm/CourtForm.tsx:88]: Maybe delete unused line?

File: src/pages/_app.tsx [hunk 9]
```diff
@@ -1,18 +1,20 @@
 import React from "react";
 import { AppProps } from "next/app";
-import { Provider } from "react-redux";
-import store from "../store";
+// import { Provider } from "react-redux";
+// import store from "../store";
```
Comment (Jzeva) [inline src/pages/_app.tsx:4]: Same here

File: src/pages/courts/[courtID]/index.tsx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  useEffect(() => {
+    api(apiUrl as string, { method: "get" }).then(({ data }) => {
+      ["_id", "isDeleted"].forEach((e) => delete data[e]);
+      delete data.isDeleted;
+      data["createdAt"] = formatDate(data.createdAt);
+      data["updatedAt"] = formatDate(data.createdAt);
+      setCourtData(data);
+      setLoading(false);
+    });
+  }, []);
```
Comment (Jzeva) [inline src/pages/courts/[courtID]/index.tsx:38]: It won't be so much different for function execution, but maybe using "async...await"  would be more consistent with other async function.

**Review by Jzeva — APPROVED** [review]

**Pull request closed without merging**